aio_pika/queue.py

```diff
@@ -398,18 +398,11 @@ async def close(self, *_: Any) -> Any:
 
         log.debug("Queue iterator %r closed", self)
 
-        def queue_tail(
-            channel: aiormq.abc.AbstractChannel,
-        ) -> Generator[Any, AbstractIncomingMessage, None]:
-            while not channel.is_closed:
-                with contextlib.suppress(asyncio.QueueEmpty):
-                    yield self._queue.get_nowait()
-                return None
-
         # Reject all messages
-        msg: IncomingMessage
-        for msg in queue_tail(self._amqp_queue.channel):
-            await msg.reject(requeue=True)
+        with contextlib.suppress(asyncio.QueueEmpty):
+            while not self._amqp_queue.channel.is_closed:
+                msg: IncomingMessage = self._queue.get_nowait()
+                await msg.reject(requeue=True)
```

Comment:
I will be really happy if you try my method with `.nack(requeue=True, multiple=True)` only for the last message.

Reply:
Done